### PR TITLE
Support new Index type IVFPQ #127

### PR DESCRIPTION
**Pull request by Yukikaze-CZR**
#127

### CODE REVIEW

**Comment by JinHai-CN** [comment]
Compiling failed @Yukikaze-CZR

**Review by tinkerlin — APPROVED** [review]
nice

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,21 +4,31 @@ Please mark all change in change log and use the ticket from JIRA.
 # Milvus 0.6.0 (TODO)
 
 ## Bug
+
 - \#246 - Exclude src/external folder from code coverage for jenkin ci
 - \#248 - Reside src/external in thirdparty
 
+##
```
Comment (JinHai-CN) [inline CHANGELOG.md]: remove these unused tags

File: CHANGELOG.md [hunk 2]
```diff
@@ -368,11 +378,11 @@ Please mark all change in change log and use the ticket from JIRA.
 - MS-82 - Update server startup welcome message
 - MS-83 - Update vecwise to Milvus
 - MS-77 - Performance issue of post-search action
-- MS-22 - Enhancement for MemVector size control
+- MS-22 - Enhancement for MemVector size control
```
Comment (JinHai-CN) [inline CHANGELOG.md]: Remove the space

File: core/conf/server_gpu_config.template [hunk 3]
```diff
@@ -29,7 +29,7 @@ metric_config:
 cache_config:
   cpu_cache_capacity: 16            # GB, CPU memory used for cache, must be a positive integer
   cpu_cache_threshold: 0.85         # percentage of data that will be kept when cache cleanup is triggered, must be in range (0.0, 1.0]
-  gpu_cache_capacity: 4             # GB, GPU memory used for cache, must be a positive integer
+  gpu_cache_capacity: 3             # GB, GPU memory used for cache, must be a positive integer
```
Comment (JinHai-CN) [inline core/conf/server_gpu_config.template]: Why change it from 4 to 3?

File: core/src/wrapper/ConfAdapter.cpp [hunk 4]
```diff
@@ -87,7 +87,11 @@ knowhere::Config
 IVFConfAdapter::MatchSearch(const TempMetaConf& metaconf, const IndexType& type) {
     auto conf = std::make_shared<knowhere::IVFCfg>();
     conf->k = metaconf.k;
-    conf->nprobe = metaconf.nprobe;
+
+    if (metaconf.nprobe <= 0)
+        conf->nprobe = 16;  // hardcode here
```
Comment (JinHai-CN) [inline core/src/wrapper/ConfAdapter.cpp]: Why use 16?

File: core/src/wrapper/ConfAdapter.cpp [hunk 5]
```diff
@@ -123,11 +127,45 @@ IVFPQConfAdapter::Match(const TempMetaConf& metaconf) {
...
     return conf;
 }
 
+knowhere::Config
+IVFPQConfAdapter::MatchSearch(const TempMetaConf& metaconf, const IndexType& type) {
+    auto conf = std::make_shared<knowhere::IVFPQCfg>();
+    conf->k = metaconf.k;
+
+    if (metaconf.nprobe <= 0)
+        conf->nprobe = 16;  // hardcode here
```
Comment (JinHai-CN) [inline core/src/wrapper/ConfAdapter.cpp]: Why this value?

File: core/src/wrapper/VecIndex.h [hunk 6]
```diff
@@ -47,6 +48,7 @@ enum class IndexType {
     FAISS_IVFSQ8_GPU,
     FAISS_IVFSQ8_HYBRID,  // only support build on gpu.
     NSG_MIX,
+    FAISS_IVFPQ_MIX,
```
Comment (JinHai-CN) [inline core/src/wrapper/VecIndex.h]: No IVFPQ_CPU/IVFPQ_GPU?

File: core/unittest/wrapper/utils.h [hunk 7]
```diff
@@ -15,18 +15,21 @@
...
 #include <gtest/gtest.h>
-#include <cstdio>
-#include <cstdlib>
-#include <fstream>
 #include <memory>
 #include <vector>
+#include <cstdlib>
+#include <cstdio>
+#include <fstream>
+#include <src/wrapper/ConfAdapter.h>
```
Comment (JinHai-CN) [inline core/unittest/wrapper/utils.h]: src shouldn't appear the path

File: core/src/wrapper/ConfAdapter.cpp (hunk 5 above)
Comment (tinkerlin) [reply core/src/wrapper/ConfAdapter.cpp]: keey accuracy

**Pull request merged**